stl/inc/deque

```diff
@@ -722,43 +722,37 @@ public:
         _Take_contents(_Right);
     }
 
-private:
-    void _Move_assign(deque& _Right, _Equal_allocators) noexcept {
-        _Tidy();
-        _Pocma(_Getal(), _Right._Getal());
-        _Take_contents(_Right);
-    }
-
-    void _Move_assign(deque& _Right, _Propagate_allocators) {
-        auto& _Al       = _Getal();
-        auto& _Right_al = _Right._Getal();
-        if (_Al == _Right_al) {
-            _Move_assign(_Right, _Equal_allocators{});
-        } else {
-            _Alproxy_ty _Alproxy(_Al);
-            _Alproxy_ty _Right_alproxy(_Right_al);
-            _Container_proxy_ptr12<_Alproxy_ty> _Proxy(_Right_alproxy, _Leave_proxy_unbound{});
-            _Tidy();
-            _Pocma(_Al, _Right_al);
-            _Proxy._Bind(_Alproxy, _STD addressof(_Get_data()));
-            _Take_contents(_Right);
+public:
+    deque& operator=(deque&& _Right) noexcept(_Alty_traits::is_always_equal::value) {
+        if (this == _STD addressof(_Right)) {
+            return *this;
         }
-    }
 
-    void _Move_assign(deque& _Right, _No_propagate_allocators) {
-        if (_Getal() == _Right._Getal()) {
-            _Move_assign(_Right, _Equal_allocators{});
-        } else {
-            assign(
-                _STD make_move_iterator(_Right._Unchecked_begin()), _STD make_move_iterator(_Right._Unchecked_end()));
+        constexpr auto _Pocma_val = _Choose_pocma_v<_Alty>;
+        if constexpr (_Pocma_val == _Pocma_values::_Propagate_allocators) {
+            auto& _Al       = _Getal();
+            auto& _Right_al = _Right._Getal();
+            if (_Al != _Right_al) {
+                _Alproxy_ty _Alproxy(_Al);
+                _Alproxy_ty _Right_alproxy(_Right_al);
+                _Container_proxy_ptr12<_Alproxy_ty> _Proxy(_Right_alproxy, _Leave_proxy_unbound{});

```

Comment:
I am wondering why this is not using `_Reload_proxy` like the other containers

Reply:
It uses `_Reload_proxy` for copy assignments.
`basic_string`, conversely, does not use `_Reload_proxy` for move assignments.
I don't know the reason, currently the change is mechanical, without doing much besides tag dispatch removal.

Reply:
Having video reviewed this, I would like to thank you profusely for keeping this PR purely as a conversion from tag dispatch to `if constexpr` =)